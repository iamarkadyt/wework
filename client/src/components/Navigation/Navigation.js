import React from 'react'
import './Navigation.scss'
import { connect } from 'react-redux'
import { logoutUser } from '../../state/actions/userActions'
import { withRouter } from 'react-router-dom'
import Dropdown from './Dropdown/Dropdown'
import ProtectedLinks from './ProtectedLinks/ProtectedLinks'
import ProfileButton from './ProfileButton/ProfileButton'

class Navigation extends React.Component {
  state = {
    showDropdown: false
  }

  toggleDropdown(e, state) {
    this.setState(prevState => ({ 
      showDropdown: state || !prevState.showDropdown
    }))
  }

  render() {
    const { 
      history, 
      logoutUser,
      authedUser
    } = this.props

    return <div className="Navigation-container">
      <div className="Navigation-logo">
        <span>We</span><span>Work</span>
      </div>
      <nav>
        <a 
          href="https://arkadyt.com"
          className="Navigation-semi-transparent"
          target="_blank" rel="noopener noreferrer">
          App Author
        </a>
        <ProtectedLinks 
          isAuthenticated={authedUser.isAuthenticated} />
        <ProfileButton 
          toggleDropdown={this.toggleDropdown.bind(this)}
          isAuthenticated={authedUser.isAuthenticated} />
        <Dropdown 
          showDropdown={this.state.showDropdown}
          toggleDropdown={this.toggleDropdown.bind(this)}
          history={history}
          logoutUser={logoutUser} />
      </nav>
    </div>
  }
}

export { Navigation as _UnconnectedNavigation }
export default withRouter(connect(state => ({
  authedUser: state.user
}), { logoutUser  })(Navigation))
