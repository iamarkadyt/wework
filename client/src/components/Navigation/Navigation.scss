@import '_variables.scss';
.Navigation-container {
    width: 100%;
    height: $navigation_height;
    background-color: #212121;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 10;
    nav {
        float: right;
        display: flex;
        height: 100%;
        margin-right: 1rem;
        a {
            color: white;
            height: 100%;
            display: flex;
            align-items: center;
            text-decoration: none;
            &:hover {
                background-color: #414141;
            }
            padding: 0 1rem;
        }
        .account-menu {
            position: fixed;
            top: calc(#{$navigation_height} + .5rem);
            right: .5rem;
            width: 10rem;
            list-style: none;
            background-color: #212121;
            overflow: hidden;
            border-radius: .3rem;
            z-index: 10;
            li>a {
                height: 3rem;
                text-transform: uppercase;
                font-weight: bolder;
            }
        }
        .backdrop {
            position: fixed;
            top: 0;
            left: 0;
            width: 100vw;
            height: 100vh;
        }
    }
}