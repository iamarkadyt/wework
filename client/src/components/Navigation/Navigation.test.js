import React from 'react'
import { shallow } from 'enzyme'
import { _UnconnectedNavigation as Navigation } from './Navigation'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    logoutUser: jest.fn(cb => cb()),
    authedUser: { isAuthenticated: true },
    history: { push: jest.fn() }
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Navigation {...props} />)
  }
  return mountedComponent
}

describe('Navigation', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('passing', () => {
      describe('ProfileButton', () => {
        it('receives correct toggleDropdown prop', () => {
          expect(comp().find('profileButton').prop('toggleDropdown').name).toContain('toggleDropdown')
        })
      })

      describe('Dropdown', () => {
        it('receives state.showDropdown into showDropdown prop', () => {
          comp().setState({ showDropdown: true })
          expect(comp().find('dropdown').prop('showDropdown')).toBe(comp().state('showDropdown'))
        })

        it('receives undefined into showDropdown prop if state.showDropdown is not defined', () => {
          comp().setState({ showDropdown: undefined })
          expect(comp().find('dropdown').prop('showDropdown')).not.toBeDefined()
        })

        it('receives correct function into toggleDropdown prop', () => {
          expect(comp().find('dropdown').prop('toggleDropdown').name).toContain('toggleDropdown')
        })
      })
    })
  })

  describe('functions', () => {
    describe('toggleDropdown', () => {
      it('calls setState', () => {
        const mock = jest.spyOn(Navigation.prototype, 'setState')
        comp().instance().toggleDropdown()
        expect(mock).toHaveBeenCalled()
      })

      it('calls setState with correct args', () => {
        const mock = jest.spyOn(Navigation.prototype, 'setState')
        comp().instance().toggleDropdown({}, true)
        expect(mock).toHaveBeenCalledWith({ showDropdown: true })
      })
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})
