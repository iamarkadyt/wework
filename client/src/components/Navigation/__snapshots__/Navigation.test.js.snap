// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<div
  className="Navigation-container"
>
  <div
    className="Navigation-logo"
  >
    <span>
      We
    </span>
    <span>
      Work
    </span>
  </div>
  <nav>
    <a
      className="Navigation-semi-transparent"
      href="https://arkadyt.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      App Author
    </a>
    <protectedLinks
      isAuthenticated={true}
    />
    <profileButton
      isAuthenticated={true}
      toggleDropdown={[Function]}
    />
    <dropdown
      history={
        Object {
          "push": [MockFunction],
        }
      }
      logoutUser={[MockFunction]}
      showDropdown={false}
      toggleDropdown={[Function]}
    />
  </nav>
</div>
`;
